### PR TITLE
Pipe 409 published fabs schema create

### PR DESCRIPTION
**Pull request by bkschmidt**
**Description:**
Define Delta Lake Table Schema for raw.published_fabs based on Broker Postgres published_fabs / USAS Postgres source_assistance_transaction
**Technical details:**
Define Delta Lake Table Schema for raw.published_fabs based on Broker Postgres published_fabs / USAS Postgres source_assistance_transaction
Define a TABLE_SPEC for published_fabs that references the SQL script from step #1. This can be defined directly in the create_delta_table command for now. It may be moved during the implementation of https://federal-spending-transparency.atlassian.net/browse/PIPE-183 
Ensure that the TABLE_SPEC has all the required fields to be runnable via the create_delta_table and load_table_to_delta commands.
Create a unit test to test the table creation.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Frontend <OPTIONAL>
    - [x] Operations <OPTIONAL>
    - [x] Domain Expert <OPTIONAL>
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [PIPE-409](https://federal-spending-transparency.atlassian.net/browse/PIPE-409):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/transactions/delta_models/published_fabs.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    "award_modification_amendme": {"delta": "STRING", "postgres": "TEXT"},
+    "awardee_or_recipient_legal": {"delta": "STRING", "postgres": "TEXT"},
+    "awardee_or_recipient_uniqu": {"delta": "STRING", "postgres": "TEXT"},
+    "awarding_agency_code": {"delta": "STRING", "postgres": "TEXT"},
+    "awarding_agency_name": {"delta": "STRING", "postgres": "TEXT"},
+    "awarding_office_code": {"delta": "STRING", "postgres": "TEXT"},
+    "awarding_office_name": {"delta": "STRING", "postgres": "TEXT"},
+    "awarding_sub_tier_agency_c": {"delta": "STRING", "postgres": "TEXT"},
+    "awarding_sub_tier_agency_n": {"delta": "STRING", "postgres": "TEXT"},
+    "business_categories": {"delta": "ARRAY<STRING>", "postgres": "ARRAY[]TEXT"},
```
Comment (collinwr) [inline usaspending_api/transactions/delta_models/published_fabs.py:19]: I believe the Postgres type here should be `TEXT[]`

File: usaspending_api/transactions/delta_models/published_fabs.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    "business_funds_ind_desc": {"delta": "STRING", "postgres": "TEXT"},
+    "business_funds_indicator": {"delta": "STRING", "postgres": "TEXT"},
+    "business_types": {"delta": "STRING", "postgres": "TEXT"},
+    "business_types_desc": {"delta": "STRING", "postgres": "TEXT"},
+    "cfda_number": {"delta": "STRING", "postgres": "TEXT"},
+    "cfda_title": {"delta": "STRING", "postgres": "TEXT"},
+    "correction_delete_ind_desc": {"delta": "STRING", "postgres": "TEXT"},
+    "correction_delete_indicatr": {"delta": "STRING", "postgres": "TEXT"},
+    "created_at": {"delta": "TIMESTAMP", "postgres": "TIMESTAMP"},
+    "face_value_loan_guarantee": {"delta": "NUMERIC(23,2)", "postgres": "NUMERIC"},
```
Comment (collinwr) [inline usaspending_api/transactions/delta_models/published_fabs.py:29]: On our other models, we have `NUMERIC(23,2)` for postgres on our `NUMERIC` types

File: usaspending_api/transactions/delta_models/published_fabs.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
+PUBLISHED_FABS_COLUMNS = {
+    "published_fabs_id": {"delta": "INTEGER", "postgres": "UUID"},
```
Comment (collinwr) [inline usaspending_api/transactions/delta_models/published_fabs.py:2]: Any reason you're converting this to a `UUID` here?

File: usaspending_api/etl/management/commands/load_table_to_delta.py [hunk 4]
```diff
@@ -102,6 +105,23 @@
...
     },
+    "published_fabs": {
+        "model": SourceAssistanceTransaction,
+        "is_from_broker": False,
+        "source_table": "published_fabs",
+        "source_database": "raw",
+        "destination_database": "raw",
+        "swap_table": None,
+        "swap_schema": None,
+        "partition_column": "detached_award_procurement_id",
```
Comment (collinwr) [inline usaspending_api/etl/management/commands/load_table_to_delta.py:116]: I believe this should be `published_fabs_id`

File: usaspending_api/etl/management/commands/load_table_to_delta.py [hunk 5]
```diff
@@ -102,6 +105,23 @@
         "column_names": list(TRANSACTION_FABS_COLUMNS),
         "tsvectors": None,
     },
+    "published_fabs": {
+        "model": SourceAssistanceTransaction,
+        "is_from_broker": False,
+        "source_table": "published_fabs",
```
Comment (collinwr) [inline usaspending_api/etl/management/commands/load_table_to_delta.py:111]: The USAS table is `source_assistance_transaction`

File: usaspending_api/etl/management/commands/load_table_to_delta.py (hunk 4 above)
Comment (bkschmidt) [reply usaspending_api/etl/management/commands/load_table_to_delta.py]: addressed

File: usaspending_api/transactions/delta_models/published_fabs.py (hunk 1 above)
Comment (bkschmidt) [reply usaspending_api/transactions/delta_models/published_fabs.py]: addressed

File: usaspending_api/transactions/delta_models/published_fabs.py (hunk 2 above)
Comment (bkschmidt) [reply usaspending_api/transactions/delta_models/published_fabs.py]: addressed

File: usaspending_api/transactions/delta_models/published_fabs.py (hunk 3 above)
Comment (bkschmidt) [reply usaspending_api/transactions/delta_models/published_fabs.py]: addressed

File: usaspending_api/etl/management/commands/load_table_to_delta.py (hunk 5 above)
Comment (bkschmidt) [reply usaspending_api/etl/management/commands/load_table_to_delta.py]: addressed

File: usaspending_api/transactions/delta_models/published_fabs.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    USING DELTA
+    LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
+"""
+
+published_fabs_load_sql_string = fr"""
+    INSERT OVERWRITE {{DESTINATION_DATABASE}}.{{DESTINATION_TABLE}}
+    (
+        {",".join([col for col in PUBLISHED_FABS_DELTA_COLUMNS])}
+    )
+"""
```
Comment (dpb-bah) [inline usaspending_api/transactions/delta_models/published_fabs.py:120]: Same as the other PR, no need to include it if not used.

**Review by dpb-bah — CHANGES_REQUESTED** [review]

File: usaspending_api/transactions/delta_models/published_fabs.py (hunk 6 above)
Comment (bkschmidt) [reply usaspending_api/transactions/delta_models/published_fabs.py]: removed

**Review by dpb-bah — APPROVED** [review]

**Pull request merged**